src/features/world/ui/SpecialEventModalContent.tsx

```diff
@@ -49,38 +54,42 @@ export const Dialogue: React.FC<{
 const CONTENT_HEIGHT = 350;
 const TWENTY_FOUR_HOURS = 24 * 60 * 60 * 1000;
 
-export const SpecialEventBumpkin: React.FC<{ onClose: () => void }> = ({
-  onClose,
-}) => {
+const RequiresWallet: React.FC<{ requiresWallet: boolean }> = ({
+  requiresWallet,
+  children,
+}) =>
+  requiresWallet ? (
+    <GameWallet action="specialEvent">{children}</GameWallet>
```

Comment:
It isn't quite clear why the wallet list pops up when clicking on the NPC.

Consider adding a UI step before hand that explains why they need to connect a wallet.

"Welcome to the Earn Alliance Special Event.

To be eligible for the Earn Alliance airdrop, you must first connect your wallet"